src/de/seq.rs

```diff
@@ -1,18 +1,74 @@
 use crate::de::{DeError, DeEvent, Deserializer, XmlRead};
 use crate::events::BytesStart;
+use crate::reader::Decoder;
 use serde::de::{self, DeserializeSeed};
+#[cfg(not(feature = "encoding"))]
+use std::borrow::Cow;
 
+/// Check if tag `start` is included in the `fields` list. `decoder` is used to
+/// get a string representation of a tag.
+///
+/// Returns `true`, if `start` is not in the `fields` list and `false` otherwise.
+pub fn is_unknown(
+    fields: &'static [&'static str],
+    start: &BytesStart,
+    decoder: Decoder,
+) -> Result<bool, DeError> {
+    #[cfg(not(feature = "encoding"))]
+    let tag = Cow::Borrowed(decoder.decode(start.name())?);
+
+    #[cfg(feature = "encoding")]
+    let tag = decoder.decode(start.name());
+
+    Ok(fields.iter().all(|&field| field != tag.as_ref()))
+}
+
+/// A filter that determines, what tags should form a sequence.
+///
+/// There is a two variant of sequences:
+/// - sequence where each element represented by tags with the same name
+/// - sequence where each element can have a different tag
+///
+/// The first variant could represent a collection of structs, the second --
+/// a collection of enums.
+///
+/// In the second case we don't know what name sequence element will accept,
+/// so we take an any element. But because in XML sequences a flattened into a
+/// maps, then we could take an elements that have their own dedicated fields
+/// in a struct. To prevent this we use an `Exclude` filter, that filters out
+/// any known names of a struct fields.
```

Comment:
```suggestion
/// tags with any known field names from the struct.
```

Reply:
Then

> /// In the second case we don't know what tag name should be expected as a
> /// sequence element, so we accept any element. Since the sequence are flattened
> /// into maps, we _skip_ elements which have dedicated fields in a struct by using an 
> /// `Exclude` filter that filters out elements with names matching field names from the struct.

This describes that fact, that struct
```rust
struct AnyName {
  field: (),
  list: Vec<()>,
}
```
in XML would be represented as
```xml
<any-tag>
  <field/>
  <list/>
  <list/>
<any-tag>
```
The JSON equivalent would be
```json
{
  "field": null,
  "list": null,
  "list": null
}
```
instead of correct
```json
{
  "field": null,
  "list": [null, null]
}
```
Because collection element could be an `enum` and enums in XML could be represented as
```xml
<variant-name .../>
```
and because the type of a sequence element is opacue to us at the `AnyName` deserialization level, we should consider all tags as a potential list items with one exception: if we are know with guarantee that a tag with that name is mapped to another field. We list all names of that tags in the `Exclude` variant and this list is a `fields` list of a struct.
In order to achieve correct deserialization we should collect all tags into a sequence

Reply:
Makes sense.  I always thought the most natural XML representation was akin to the JSON one, e.g.

```xml
<any-tag>
  <field/>
  <list>
    <item/>
    <item/>
    <item/>
  </list>
<any-tag>
```

But I can see some rationale for not doing it that way, too.

Reply:
Yes, it would be nice for this to be the case, but I often see such XML. If you think about it, this is natural, given how easy it is to describe such a structure in XSD:
```xml
<xs:complexType>
  <xs:sequence>
    <xs:element name="field"/>
    <xs:element name="list" maxOccurs="unbounded"/>
  </xs:sequence>
</xs:complexType>
```